src/Compilers/Core/Portable/Serialization/StreamObjectReader.cs

```diff
@@ -0,0 +1,1020 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Utilities
+{
+    using EncodingKind = StreamObjectWriter.EncodingKind;
+    using Variant = StreamObjectWriter.Variant;
+    using VariantKind = StreamObjectWriter.VariantKind;
+
+    /// <summary>
+    /// A class that deserializes objects from a stream.
+    /// </summary>
+    internal sealed partial class StreamObjectReader : ObjectReader, IDisposable
+    {
+        private readonly BinaryReader _reader;              // reads primitives from stream
```

Comment:
I disagree.  People will need to modify the class.  In such situations features our tooling will not show the comments.

We do not gain from this, and it actually hurts the editing and maintenance of this code.  It's also a pattern we discourage elsewhere in our code on the IDE side of things.